### DISCUSSION TITLE
Kann dem Map Marker keinen Icon hinzufügen

### DISCUSSION THREAD

**Discussion by MiejkeHoltermann** [neuefische/questions#694] [Web] [answered]
Hallo,
ich habe mit leaflet und react-leaflet einen Container implementiert, indem meine entsprechende Destination auf der Open Street Map angezeigt wird. Um Probleme mit dem SSR zu umgehen, habe ich MapContainer, TileLayer und Marker dynamisch importiert, was auch so weit funktioniert hat. Allerdings wird klassischerweise für den Marker ja kein Icon angezeigt. Versuche ich diesen auf die gleiche Weise zu importieren (dynamisch) und mit folgendem Code zu implementieren

 const customIcon = new DynamicL.Icon({
    iconUrl: "map-marker.svg",
    iconSize: [50, 50],
  });

erhalte ich die Fehlermeldung
:
TypeError: DynamicL.Icon is not a constructor

Ich habe meinen aktuellen Stand committed und gepusht: https://github.com/MiejkeHoltermann/capstone-project
Der Branch heißt feat/map-view

Wie kann man den Fehler reproduzieren?
Einfach die URL um /map ergänzen, dann taucht schon die Fehlermeldung auf.

Wo und wann tritt er auf?
Sobald man versucht die Seite "map" zu laden.

Was habe ich bereits ausprobiert:
- L normal zu importieren (nicht dynamisch)
- Alle Leaflet Sachen in einer separaten Komponente Map normal zu importieren und dann diese Komponente dynamisch auf meiner Seite zu importieren
- Verschiedene Syntaxideen (z. B. von StackOverflow)
- Leaflet Documentation
- react-leaflet Documentation

Vielen Dank für eure Hilfe!

**Comment by F-Kirchhoff** [comment]
Hey Miejke,
woher kommt denn das dynamicL ? hast du versucht das "new" einmal wegzulassen oder dir auszugeben was es genau ist? Es scheint dort vllt ein Problem zu geben mit dem import von dynamicL.

**Reply by MiejkeHoltermann** [reply]
Hallo Felix,
new L.Icon wird von leaflet so vorgegeben. Wenn ich console.log(L) oder console.log(typeof L) eingebe kommt gar keine Ausgabe, nichtmal undefined.

**Reply by MiejkeHoltermann** [reply]
Das Komische ist, new L.Icon funktioniert auch, wenn ich ganz normal import L from "leaflet" schreibe und nicht dynamisch importiere. Allerdings bekomme ich dann beim Neuladen wieder den Fehler "window not defined", der durch das SSR hervorgerufen wird und den ich eigentlich durch die dynamischen Imports behoben hatte.

**Reply by F-Kirchhoff** [reply]
I can't find the lines of code that you are citing, please create a PR with the relevant code and post a link here.

**Reply by MiejkeHoltermann** [reply, marked as answer]
Hab's schon gefunden. Ich hatte vorher schon versucht eine separate Map-Komponente zu erstellen und nur diese dynamisch zu importieren, und hatte einen Error. Aber jetzt es funktioniert und damit ist auch das andere Problem behoben. Trotzdem vielen Dank!!
